src/inmanta/server/services/notificationservice.py

```diff
@@ -63,14 +63,22 @@ async def start(self) -> None:
     async def _cleanup(self) -> None:
         await data.Notification.clean_up_notifications()
 
-    async def compile_done(self, compile: data.Compile) -> None:
+    async def compile_done(self, compile: data.Compile, notify_failed_compile: bool = False) -> None:
         if not compile.success:
             compile_report = await data.Compile.get_report(compile_id=compile.id)
             reports = compile_report["reports"] if compile_report else []
             failed_pull_stage = next(
                 (report for report in reports if report["name"] == "Pulling updates" and report["returncode"] != 0), None
             )
-            if failed_pull_stage:
+            if notify_failed_compile:
```

Comment:
If `compile.notify_failed_compile is False` just return without doing anything?

Reply:
hmm no this is wrong, i'll change it.

Reply:
I think you may have misunderstood me here. I think what was there before was perfectly fine, and perhaps even a bit more readable with the `elif` than with the `else: if`. What I meant to say is you don't handle the case where `notify_failed_compile is False` (rather than `None`).

Reply:
I'm not sure I understand, I thought we still wanted to notify for the `failed_pull_stage` and the `compile.do_export` even if `notify_failed_compile` is false. Is that not the case?

Reply:
We want that to be the default (`notify_failed_compile is None`), but if the user requests `notify_failed_compile=False` I think it would be counterintuitive to still notify for exporting compiles.

Reply:
Ok now I see :p Indeed I agree and this makes more sense. thanks